src/ui-kit/experimental/hierarchical/hierarchical-tree/hierarchical-tree.component.ts

```diff
@@ -192,11 +214,52 @@ export class SamHierarchicalTreeComponent implements OnInit {
     }
   }
 
+
+
+
+
+
   /**
    * Calls the provided service to get the results for the girdbased on
    * the primary id of the selected   * and the filter
    */
-  private getResults() {
-    this.gridResults = this.service.getHiercarchicalById(this.selectedValue, this.filterText, this.sort);
+  private getResults(isScroll?: boolean) {
```

Comment:
@thomastighe I have a few concerns about this method. The first thing I noticed is the function is pretty "tall", but after looking, it doesn't have that much functionality.

It seems like, based on a condition, you're constructing some object to pass to the service, then calling it and subscribing. 

What concerns me is the number of subscriptions. Each time you make the call, a new subscription is being made. Those could potentially be floating around and making memory leaks. It really depends on how methods are being called and where components are being created or destroyed.

One way around it would be to use the pipe operator and only take the first item since (I think) we're calling this every time and that subscription is never getting anything else passed to it.

`this.service.getHierarchicalById(item).pipe(first()).subscribe(...)`

Another answer is how they do it in this stackoverflow answer (Edit 3): https://stackoverflow.com/a/41177163

Since we aren't using subjects in the service to call next, there's probably no point in this. I think taking the first is fine.

Can you also add an error handler on the subscriptions? What would happen if the service returned an error, how should the component behave? We might need some design work on that experience, but minimally, I would try to catch the error so it doesn't break the overall experience.

There isn't a corresponding else conditional if isScroll is true, but the result length <= totalItems. Some kind of debug statement here might be helpful.

Last thing -- and this is an observation and comment, not a request for changes -- it seems like there is an opportunity to write more DRY code here. There's some duplication around assignment  and creating the subscription.

However,  I think it is probably more clear to leave it the way you have it so we can get a better understanding of the logical flow. Otherwise, you could have one assignment to item and one subscription, but you'd need conditionals for the item field assignments and the subscription body.

Whatever you decide, I'm good with it.

Reply:
Made the pipe(first()) change as asked